Design.md

```diff
@@ -174,6 +174,15 @@ Originally discussed in [#12](https://github.com/coreos/fedora-coreos-tracker/is
  - For the short term, if we need to include an agent we will bake it into the image. We will not have any specific
    mechanism for including agents.
 
+### Azure:
+
+Originally discussed in [#65](https://github.com/coreos/fedora-coreos-tracker/issues/65).
+
+- We've identified one major gap with not shipping the [Microsoft Azure Linux Agent](https://github.com/Azure/WALinuxAgent): the machine will not check-in and will eventually be culled by Azure for being stuck in the creation process.
+- This gap will be temporarily worked around via a [systemd unit](https://github.com/coreos/fedora-coreos-tracker/issues/65#issuecomment-434449560) which will perform the check-in.
```

Comment:
maybe we shouldn't mention the temporary workaround since hopefully we get the coreos-metadata issue fixed sooner than later. I think @jlebon is going to start working on it soon.

WDYT?

Reply:
I would say so unless it's a much larger work item than I'm currently assuming. You know much more about it than I do so maybe you can clue me in.

Reply:
Overall an initial implementation that doesn't tackle the DHCP option parsing is very straightforward and still covers most use cases of Azure (only missing ASM [legacy] & AzureStack machines).